tree/treeplayer/src/TSimpleAnalysis.cxx

```diff
@@ -193,15 +196,32 @@ static std::string ExtractTreeName(std::string& firstInputFile)
 }
 
 ////////////////////////////////////////////////////////////////////////////////
-/// Execute all the TChain::Draw() as configured and stores the output histograms.
-/// Returns true if the analysis succeeds.
+/// Returns true if there are no errors in TChain::LoadTree()
 
-bool TSimpleAnalysis::Run()
+static bool checkChainLoadResult(TChain* chain, int& errValue, std::string& errFile)
+{
+   TObjArray *fileElements = chain->GetListOfFiles();
+   TIter next(fileElements);
+   while (TChainElement* chEl = (TChainElement*)next()) {
+      if (chEl->GetLoadResult() < 0) {
+         errValue = chEl->GetLoadResult();
```

Comment:
Instead of returning name, value - can we call Error() here (the function will have to become a member function for that). And don't return out of the loop, but check also the subsequent `TChainElement`s, and then at the end of the function return `false` if there was an error. I.e. you'll need a `bool ret = true` that you set to `false` if a load result is <0.